packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -734,6 +739,79 @@ export class ProjectService extends BaseService {
                 throw new SnowflakeTokenError(errorMessage);
             }
         }
+
+        if (
+            args.type === WarehouseTypes.DATABRICKS &&
+            args.authenticationType === DatabricksAuthenticationType.OAUTH_M2M
```

Comment:
M2M (machine to machine) uses different arguments for authentication using oauth (client/secret) and it is ocnfigured differently on databricks. 

When we introduce U2M (for require user credentials), we'll add a new authenticationType